### PR TITLE
Add install-cni test 

### PR DESCRIPTION
**Pull request by tiswanso**
This is a combo implementation to establish a couple of test cases for the install-cni container.  It currently requires docker.  

Details:
* `make test` or `make install-test`
* docker runs install-cni with dirs mounted and env vars set
   * checks the resulting CNI plugin configuration and temp bin dir contents
   * does a straight file compare of plugin config
   * checks cleanup as well

### CODE REVIEW

File: deployments/kubernetes/install/test/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+```
+### Description
+
+1. invoke container standalone in raw docker with
+   1. test temp dirs mounted to locations for CNI conf/bin & k8s secrets
+   1. env vars set like K8s would
+   1. mount test script dir
+1. Test script
+   1. `docker run install-cni` with different env var settings
+      1. test conf form format files
```
Comment (john-a-joyce) [inline deployments/kubernetes/install/test/README.md]: "form" -> from

File: deployments/kubernetes/install/test/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+         1. with pre-existing .plugins[] (10-calico.conflist)
+         1. without pre-existing .plugins[] (minikube_cni.conf)
+      1. test cleanup on sigterm
+   1. for each case do `cmp -s resultfile expectedfile`
+      1. Pass if result file != expected file
```
Comment (baodongli) [inline deployments/kubernetes/install/test/README.md]: should the result file be equal to the expected file?

File: deployments/kubernetes/install/test/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+## Test install-cni.sh
+
+### Run
+```
+./test-install-cni.sh
+```
+### Description
+
+1. invoke container standalone in raw docker with
+   1. test temp dirs mounted to locations for CNI conf/bin & k8s secrets
```
Comment (baodongli) [inline deployments/kubernetes/install/test/README.md]: for CNI conf and bin directory.

File: deployments/kubernetes/install/test/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+   1. env vars set like K8s would
+   1. mount test script dir
+1. Test script
+   1. `docker run install-cni` with different env var settings
+      1. test conf form format files
+         1. with pre-existing .plugins[] (10-calico.conflist)
+         1. without pre-existing .plugins[] (minikube_cni.conf)
```
Comment (baodongli) [inline deployments/kubernetes/install/test/README.md]: with pre-existing plugins
without pre-existing plugins

File: deployments/kubernetes/install/test/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+## Test install-cni.sh
+
+### Run
+```
+./test-install-cni.sh
+```
+### Description
+
+1. invoke container standalone in raw docker with
```
Comment (baodongli) [inline deployments/kubernetes/install/test/README.md]: container standalone?
raw docker?

**Comment by tiswanso** [comment]
Thanks, @baodongli !  I'll merge this and update the readme separately.

**Pull request merged**